dao/src/main/java/se/fortnox/reactivewizard/db/statement/StatementDebug.java

```diff
@@ -0,0 +1,27 @@
+package se.fortnox.reactivewizard.db.statement;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.PreparedStatement;
+
+public class StatementDebug {
+    private static final Logger LOG = LoggerFactory.getLogger(StatementDebug.class);
+    static boolean ENABLE_STATEMENT_DEBUG = Boolean.parseBoolean(System.getenv("dao.debug"));
+
+    private StatementDebug() {
+    }
+
+    /**
+     * Logging statement if dao.debug is set to true.
+     * This only works if an actual postgres database
+     * is running.
+     * @param statement a sql PreparedStatement
+     */
+    public static void log(PreparedStatement statement) {
+        if (ENABLE_STATEMENT_DEBUG) {
```

Comment:
Do you know if there is a reason for using an environment variable here? 

Otherwise, I think it would be easier to go with the standard way by configuring the logging through the admin port on demand or in the log4j.xml file.

Also Log4j2 support lazy logging with lambdas like this:

`LOG.debug("Executed statement: {}", () -> statement);`

If you do not want to create the string "Executed statement: {}" every time due to optimization you can put it in a final static field in the class as well.

Reply:
Making it dynamic means you would get a penalty on each call because the flag needs to be checked. If it can be set using static+final then the JVM can completely remove this code when it is compiled.

Reply:
It is not meant to be an environment variable, it should be a java property, my bad, will fix.